### PR TITLE
DOC Add more examples to Series.Resample docstring

### PR DESCRIPTION
**Pull request by Winterflower**
Partial fix for https://github.com/pydata/pandas/issues/5023
flake8 doesn't do a very good job at checking code in the docstrings for PEP8 compliance, so I'm just eyeballing it at the moment and my PEP8 knowledge leaves something to be desired. Apologies if I missed something really obvious.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -3274,7 +3274,50 @@ def resample(self, rule, how=None, axis=0, fill_method=None,
...
             range from 0 through 4. Defaults to 0
+
+        Examples
+        --------
+        
+        Start by creating a series with 9 one minute timestamps.
+
+        >>> index = pd.date_range('1/1/2000', periods=9, freq='T')
+        >>> series = pd.Series(range(9), index=index)
+        >>> print series
```
Comment (sinhrks) [inline pandas/core/generic.py]: Remove ``print`` (for py3), only variable name (``series``) is better.

File: pandas/core/generic.py [hunk 2]
```diff
@@ -3274,7 +3274,50 @@ def resample(self, rule, how=None, axis=0, fill_method=None,
...
+        >>> series.resample('3T', how='sum')
+        2000-01-01 00:00:00     3
+        2000-01-01 00:03:00    12
+        2000-01-01 00:06:00    21
+        Freq: 3T, dtype: int64
+
+        Downsample the series into 3 minute bins as above, but label each
+        bin using the right edge instead of the left. 
+
+        >>> series.resample('3T', how='sum', label='left')
```
Comment (sinhrks) [inline pandas/core/generic.py]: ``label='right'``?

Comment (Winterflower) [inline pandas/core/generic.py]: @sinhrks ah yes, sloppy me! I must have correct it in the Ipython notebook I use to check these and forgotten to place it in the file. Thanks!

**Comment by Winterflower** [comment]
Will squash everything into one commit once I add an example of using a custom ``how`` method.

**Comment by Winterflower** [comment]
@jorisvandenbossche @sinhrks 
this one is done and good to go from my side!
comments/corrections welcome, of course!

File: pandas/core/generic.py [hunk 3]
```diff
@@ -3274,7 +3274,90 @@ def resample(self, rule, how=None, axis=0, fill_method=None,
...
+        >>> series.resample('3T', how='sum')
+        2000-01-01 00:00:00     3
+        2000-01-01 00:03:00    12
+        2000-01-01 00:06:00    21
+        Freq: 3T, dtype: int64
+
+        Downsample the series into 3 minute bins as above, but label each
+        bin using the right edge instead of the left. 
+
+        >>> series.resample('3T', how='sum', label='right')
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: Maybe you also add `how=right'` here as well? (I always find that in most  common cases you need to specify both)

**Comment by jorisvandenbossche** [comment]
Nice examples!

**Comment by sinhrks** [comment]
Looks nice! No further comments from me.

**Comment by jorisvandenbossche** [comment]
@Winterflower Do you have some time to update? (it's only a very small comment)

**Comment by Winterflower** [comment]
@jorisvandenbossche apologies for the delay! I will add the how='right' example tonight after work.

File: pandas/core/generic.py (hunk 3 above)
Comment (Winterflower) [inline pandas/core/generic.py]: @jorisvandenbossche I'm trying to test some examples with ``how=right`` and I am getting exceptions. What is '``how=right`` supposed to do?

Comment (jorisvandenbossche) [inline pandas/core/generic.py]: oops, sorry, I meant 'closed' instead of 'how'

Comment (Winterflower) [inline pandas/core/generic.py]: haha no worries! ``how=right`` did seem a bit weird. I'll include ``closed=right`` in the example.

**Comment by Winterflower** [comment]
Still need to fix line lengths.

**Comment by Winterflower** [comment]
At the moment, the doctest for the string is failing because of the ``custom_resampler`` function defined in the docstring

```
Trying:
    def custom_resampler(array_like):
Expecting:
            return np.sum(array_like)+5
**********************************************************************
File "pandas/core/generic.py", line 3367, in generic.NDFrame.resample
Failed example:
    def custom_resampler(array_like):
Exception raised:
    Traceback (most recent call last):
      File "/home/winterflower/anaconda/lib/python2.7/doctest.py", line 1315, in __run
        compileflags, 1) in test.globs
      File "<doctest generic.NDFrame.resample[9]>", line 1
        def custom_resampler(array_like):
                                        ^
    SyntaxError: unexpected EOF while parsing
```

**Comment by Winterflower** [comment]
This is good to go on my part!

**Comment by jreback** [comment]
@Winterflower gr8 thanks!

Further I think portions of your narrative my be a nice addition to the actual resample docs; add where appropriate

**Pull request merged**